webapp/src/main/java/com/box/l10n/mojito/smartling/SmartlingClient.java

```
@@ -0,0 +1,188 @@
+    // Authentication tokens can be refreshed for up to 24 hours
+    Integer authenticationDurationHours = 23;
+
+    public SmartlingClient(String userIdentifier, String userSecret) {
+        this.userIdentifier = userIdentifier;
+        this.userSecret = userSecret;
+    }
+
+    Map<String, Object> getPayloadMapAuthentication() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("userIdentifier", userIdentifier);
+        map.put("userSecret", userSecret);
+        return map;
+    }
+
+    Map<String, Object> getPayloadMapRefresh() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("refreshToken", refreshToken);
+        return map;
+    }
+
+    Map<String, Object> getPayloadMapPullSourceString() {
+        return new HashMap<>();
+    }
+
+    HttpHeaders getHttpHeadersForJson() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return headers;
+    }
+
+    HttpHeaders getHttpHeadersForJsonWithAuthorization() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("Authorization", "Bearer " + authToken);
+        return headers;
+    }
+
+    <T> HttpEntity<T> getHttpEntityForPayload(T payload) {
+        return new HttpEntity<>(payload, getHttpHeadersForJson());
+    }
+
+    <T> HttpEntity<T> getHttpEntityForPayloadWithAuthorization(T payload) {
+        return new HttpEntity<>(payload, getHttpHeadersForJsonWithAuthorization());
+    }
+
+    public void authenticate() throws SmartlingClientException {
+        Map<String, Object> payload = getPayloadMapAuthentication();
+        AuthenticationResponse authenticationResponse;
+        try {
+            authenticationResponse = postForAuthentication(payload, AUTH_API_V2_AUTHENTICATE);
+        } catch (SmartlingClientException e) {
+            throw new SmartlingClientException(e.getMessage());
```

here, you should just re-throw the exception, not re-create one.

```
catch (SmartlingClientException e) {
            throw e;
```